tests/services/analysis_starter/test_balsamic_configurator.py

```diff
@@ -0,0 +1,439 @@
+
+    # WHEN getting the normal sample id
+    normal_sample_id = BalsamicConfigurator._get_normal_sample_id(case_with_tumor)
+
+    # THEN the normal sample id should be None
+    assert normal_sample_id is None
+
+
+def test__get_tumor_sample_id():
+    """Tests that a case containing a tumor sample returns the correct sample id."""
+    # GIVEN a case with a tumor sample
+    case_with_tumor = create_autospec(Case)
+    tumor_sample = create_autospec(Sample, internal_id="tumor_sample", is_tumour=True)
+    case_with_tumor.samples = [tumor_sample]
+
+    # WHEN getting the tumor sample id
+    tumor_sample_id = BalsamicConfigurator._get_tumor_sample_id(case_with_tumor)
+
+    # THEN the correct tumor sample id should be returned
+    assert tumor_sample_id == "tumor_sample"
+
+
+def test__get_tumor_sample_id_no_tumor_sample():
+    """Tests that a case without tumor samples raises an error."""
+    # GIVEN a case without a tumor sample
+    case_with_normal = create_autospec(Case)
+    normal_sample = create_autospec(Sample, internal_id="normal_sample", is_tumour=False)
+    case_with_normal.samples = [normal_sample]
+
+    # WHEN getting the tumor sample id
+    # THEN it should raise a ValueError
+    with pytest.raises(BalsamicMissingTumorError):
+        BalsamicConfigurator._get_tumor_sample_id(case_with_normal)
+
+
+def test__build_cli_input_wgs_tumor_only(balsamic_configurator):
+    # GIVEN a case with one WGS sample
+    wgs_tumor_only_case = create_autospec(Case, data_analysis="balsamic", internal_id="case_1")
+    application = create_autospec(Application, prep_category="wgs")
+    application_version = create_autospec(ApplicationVersion, application=application)
+    sample_1 = create_autospec(
+        Sample,
+        internal_id="sample_1",
+        is_tumour=True,
+        sex=SexOptions.FEMALE,
+        application_version=application_version,
+    )
+    wgs_tumor_only_case.samples = [sample_1]
+    balsamic_configurator.store.get_case_by_internal_id = Mock(return_value=wgs_tumor_only_case)
+
+    # WHEN building the CLI input
+    cli_input: BalsamicConfigInput = balsamic_configurator._build_cli_input(
+        case_id=wgs_tumor_only_case.internal_id
+    )
+
+    # THEN pydantic model should be created (and validated)
+    assert isinstance(cli_input, BalsamicConfigInput)
+
+    # THEN the correct no normal sample name should be set
+    assert cli_input.normal_sample_name is None
+
+    # THEN the correct gens_coverage_pon should be chosen (Female)
+    assert cli_input.gens_coverage_pon == balsamic_configurator.gens_coverage_female_path
+
+    # THEN fields not relevant for WGS analyses should be set to their default values
+    for field in PANEL_ONLY_FIELDS:
+        assert getattr(cli_input, field) == BalsamicConfigInput.model_fields[field].default
+
+
+def test__build_cli_input_wgs_tumor_normal(balsamic_configurator):
+    # GIVEN a case with one two WGS samples (tumor and normal)
```

Comment:
```suggestion
    # GIVEN a case with two WGS samples (tumor and normal)
```

Reply:
fixed in 55a9f5ae2a98a97a78b86eee6f8e13194d48b10d